### ISSUE TITLE
[Feature]: When upgrading a package, if a transient package is also "directly" installed, upgrade it to avoid NU1605

### ISSUE THREAD

**Issue by Rizzen** [NuGet/Home#11019]
### NuGet Product Used

NuGet.exe, Visual Studio Package Management UI

### Product Version

VS 16.10

### Worked before?

No

### Impact

It bothers me. A fix would be nice

### Repro Steps & Context

- Сreate a test project targeting dotnet core (2 or higher)
- Add package **NExpect.Matchers.NSubstitute** at version 1.0.150 -> should pull in **NExpect** at 1.0.150 (ensure that you have a package reference for **NExpect** - may have to add it manually; just use the same version)
- Try to upgrade either package. **NExpect.Matchers.NSubstitute** will not upgrade because it tracks the same version of **NExpect**
- Attempting to upgrade **NExpect.Matchers.NSubstitute** fails with the error:

```
Detected package downgrade: NExpect from 1.0.162 to 1.0.150. Reference the package directly from the project to select a different version. 
NugetUpgrades -> NExpect.Matchers.NSubstitute 1.0.162 -> NExpect (>= 1.0.162) 
NugetUpgrades -> NExpect (>= 1.0.150)
```

It's old and very annoying problem and it would be nice to test the dependencies of a package to be upgraded and suggest like "Upgrading [X] to [a.b.c] requires upgrading [Y] to [e.f.g]. Continue?"

![image1](https://user-images.githubusercontent.com/23218891/125194350-8ec25800-e259-11eb-8154-db5871b1055c.png)

**Comment by heng-liu** [comment]
Hi @Rizzen, I tried to repro and it did show NU1605 if I only chose one package to upgrade.
But if I checked both the two packages and click "Updates". Then both of them are updated successfully. There is no downgrade error anymore.
I think this is fixed by https://github.com/NuGet/NuGet.Client/pull/3559.
Could you try the same approach and let us know if it works for you? Thanks!

**Comment by Rizzen** [comment]
@heng-liu Yeah, the way you describe works, but common use case is single package upgrade and it leads to NU1605. It would be nice if NuGet updated all the neccessary dependencies when upgrading single package.

**Comment by zivkan** [comment]
@Rizzen I changed this issue to a new feature request, since it's not a bug of existing features.

Also, you can consider uninstalling NExpect as a direct reference, since it's a transitive package of the other package, and let it come in that way.

**Comment by nkolev92** [comment]
Similar issue: https://developercommunity.visualstudio.com/t/NU1605-Warning-as-error-turns-the-Visual/10886751, related to https://github.com/nuget/home/issues/14011. 

The gist is that sometimes you need to update multiple packages to make the dependency graph happy NU1605 and vuln warnings are just the more common scenarios of it.
